### PR TITLE
Remove footer from source maps and eval

### PR DESCRIPTION
**Pull request by ruszki**
**What kind of change does this PR introduce?**
refactoring

**Did you add tests for your changes?**
No.

**If relevant, link to documentation update:**
N/A

**Summary**
Some development tool (eg Chrome DevTools) needs to get unmodified source codes.

Thus, SourceMapDevTool and EvalSourceMapDevTool plugins accept a new option called "noFooter". If it's true then no footer is generated for source maps.

This is a basic solution to #6400

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by ruszki** [comment]
I've added a new "noFooter" option, because my previous solution was very hacky.

File: lib/ModuleFilenameHelpers.js [hunk 1]
```diff
@@ -99,22 +99,31 @@ ModuleFilenameHelpers.createFilename = function createFilename(module, moduleFil
 		.replace(ModuleFilenameHelpers.REGEXP_HASH, hash);
 };
 
-ModuleFilenameHelpers.createFooter = function createFooter(module, requestShortener) {
+ModuleFilenameHelpers.createFooter = function createFooter(module, requestShortener, sourceUrlComment) {
```
Comment (sokra) [inline lib/ModuleFilenameHelpers.js]: The `sourceUrlComment` is not part of the footer. It doesn't make sense to move it into this module.

Comment (ruszki) [reply lib/ModuleFilenameHelpers.js]: You're right, sorry. I've made the requested change.

What was the use case which required this feature? I don't see the point, but I also don't know much about Webpack's internals.

**Comment by ruszki** [comment]
Did I make a mistake, or Travis is the faulty here? Tests are successful on my machine.

**Comment by ruszki** [comment]
What was the use case which required this feature? I don't see the point, but I also don't know much about Webpack's internals.

**Comment by ooflorent** [comment]
@ruszki CI had a random failure here.

Could you:

- rebase your work on `next`
- target `next` branch for this PR (or create another one)
- remove `ModuleFilenameHelpers.createFooter`

By doing all this, there no longer will be the footer in webpack 4.

**Pull request closed without merging**